src/btree/bt_ret.c

```diff
@@ -136,14 +138,17 @@ __wt_value_return_buf(
             stop->txnid = unpack.stop_txn;
         }
 
-        return (__wt_page_cell_data_ref(session, page, &unpack, buf));
+        if (buf != NULL)
+            return (__wt_page_cell_data_ref(session, page, &unpack, buf));
+        return (0);
```

Comment:
I'd go with `return (buf == NULL ? 0 : XXX);` in both of these cases.

Reply:
done